### PR TITLE
ENH: Speed up next_fast_len

### PR DESCRIPTION
**Pull request by peterbell10**
#### Reference issue
Closes #10804, closes #10589

#### What does this implement/fix?
This supercedes #10804, extending it to also speed up `next_fast_len.__wrapped__`. The python overhead is brought down from ~550 ns to ~85 ns by binding the function directly in the c-api and replacing the `kind` argument with the `real` argument used in the C++ binding version.

#### Additional information
Here are the improved timings I see:
```
In [1]: from scipy.fft import next_fast_len
In [2]: %timeit next_fast_len.__wrapped__(12)
86.8 ns ± 3.78 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
In [3]: %timeit next_fast_len.__wrapped__(997)
222 ns ± 3.43 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In [4]: %timeit next_fast_len.__wrapped__(1000)
135 ns ± 0.593 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
In [5]: %timeit next_fast_len.__wrapped__(16381)
415 ns ± 10.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

cc @larsoner @mreineck

### CODE REVIEW

**Comment by larsoner** [comment]
Travis failure looks legitimate:
```
=============
scipy.fftpack
=============
scipy.fftpack.next_fast_len
---------------------------
File "build/testenv/lib/python3.6/site-packages/scipy/fftpack/helper.py", line 87, in next_fast_len
Failed example:
    fftpack.helper.next_fast_len(min_len)
Exception raised:
    Traceback (most recent call last):
      File "/opt/python/3.6.7/lib/python3.6/doctest.py", line 1330, in __run
        compileflags, 1), test.globs)
      File "<doctest next_fast_len[4]>", line 1, in <module>
        fftpack.helper.next_fast_len(min_len)
      File "/home/travis/build/scipy/scipy/build/testenv/lib/python3.6/site-packages/scipy/fftpack/helper.py", line 98, in next_fast_len
        return _helper.next_fast_len(target, True)
    AttributeError: module 'scipy.fft._pocketfft.helper' has no attribute 'next_fast_len'
```

File: scipy/fft/_helper.py [hunk 1]
```diff
@@ -60,7 +59,11 @@ def next_fast_len(target, kind='C2C'):
     >>> b = fft.fft(a, 131072)
 
     """
-    return _helper.next_fast_len(target, kind)
+    pass
+
+
+next_fast_len = update_wrapper(lru_cache()(_helper.good_size), next_fast_len)
+next_fast_len.__wrapped__ = _helper.good_size
```
Comment (larsoner) [inline scipy/fft/_helper.py]: Can you add a comment about why this is necessary? I assume it's a way to get a reasonable docstring and `args` def while having the call wrap to a C routine.

**Review by larsoner — APPROVED** [review]
Nice speed increases, and better than what I was doing (incl. with the benchmark).

**Review by grlee77 — APPROVED** [review]
Looks good. Thanks

**Pull request merged**